lib/ansible/constants.py

```diff
@@ -197,5 +199,18 @@ def __getitem__(self, y):
 
     set_constant(setting.name, value)
 
+
+# deal with 'smart' connection .. one time ..
+if DEFAULT_TRANSPORT == 'smart':
+    # TODO: check if we can deprecate this as ssh w/o control persist should
+    # not be as common anymore.
+
+    # see if SSH can support ControlPersist if not use paramiko
+    if not check_for_controlpersist(ANSIBLE_SSH_EXECUTABLE) and paramiko is not None:  # pylint: disable=undefined-variable
+        DEFAULT_TRANSPORT = "paramiko"
+    else:
+        DEFAULT_TRANSPORT = "ssh"
```

Comment:
We shouldn't be checking for control persist as part of loading constants. This results in every ansible command checking for control persist, even if those commands never use SSH (`ansible --version`, `ansible-doc`, non-SSH transports, etc.)

It also means that any other code (sanity tests, etc.) that use nearly any Ansible code are performing these same checks as well, since most end up using constants.

We really need a way to check this on-demand, but we're currently lacking a good mechanism for that. That's something we can hopefully address after 2.10.

Reply:
Yeah, I'd tend to agree. Trying to think of another place to do this that lets us do it as late as possible (eg only when something actually tried to "use" `smart`), but still pre-fork and cacheable.

Reply:
yep, found spot we already tried to do this, added it there, still modifies the constant so it avoids all the jumps when using defaults. if set at a lower level, it still does the work.